### ISSUE TITLE
v2.23 broke alphabetical sorting

### ISSUE THREAD

**Issue by jeron-diovis** [benmosher/eslint-plugin-import#2064]
[This](https://github.com/benmosher/eslint-plugin-import/commit/1c10e79dc4761d69c5ce26736541f8d024ce7339#diff-2825fcc427c846fe18d501930e1a3d83be722027f1f643801afdcc2c255dd4a8R268) and [this](https://github.com/benmosher/eslint-plugin-import/commit/1c10e79dc4761d69c5ce26736541f8d024ce7339#diff-2825fcc427c846fe18d501930e1a3d83be722027f1f643801afdcc2c255dd4a8R293) changes cause the following weird effect.

With imports like this:
```js
import React from 'react'
import { BrowserRouter } from 'react-router-dom'
...
```
func `mutateRanksToAlphabetize` produces following groups:
```js
{
  '0.8': [ 'react-value', 'react-router-dom-value' ],
  ...
}
```
Then this group gets sorted alphabetically, and obviously turns into `[ 'react-router-dom-value', 'react-value' ]`. While normally, `'react' < 'react-router-dom' === true`.

So after updating to 2.23 I suddenly have warnings about wrong imports order all over the project . 

Probably, you may just use `|` instead of `-` to separate importKind from importName?

**Comment by ruscon** [comment]
+1 have the same issue with v2.23.0

**Comment by ljharb** [comment]
cc @grit96 / #2021 - could you take a look at this?

**Comment by heath-freenome** [comment]
This broke my builds too

**Comment by grit96** [comment]
Perhaps the change should be reverted for now. I will try to take a look at fixing it next week.

**Comment by ljharb** [comment]
That's a pretty major thing to revert, I'm hoping we can roll forward.

**Comment by grit96** [comment]
I've created a fix in #2071 and added a test for the code in the OP
